### PR TITLE
Enzyme: simplify via mixedduplicated

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

File: ext/EnzymeExt.jl [hunk 1]
```diff
@@ -150,7 +126,7 @@ module EnzymeExt
         args2 = ntuple(Val(N)) do i
             Base.@_inline_meta
             if args[i] isa Active
-                Duplicated(Ref(args[i].val), arg_refs[i])
+                EnzymeCore.MixedDuplicated(args[i].val, arg_refs[i])
```
Comment (vchuravy) [inline ext/EnzymeExt.jl:129]: This is correct for the CPU, but I think we won't be able to pass `MixedDuplicated` to the GPU. 
Or at least we need an Adapt rule.

Comment (wsmoses) [reply ext/EnzymeExt.jl:129]: Do we not have those already like we do for the other annotations?

Comment (wsmoses) [reply ext/EnzymeExt.jl:129]: Apparently not, now we do here: https://github.com/EnzymeAD/Enzyme.jl/pull/1551

**Comment by wsmoses** [comment]
@vchuravy in interim could we get this merged/tagged (is blocking https://github.com/CliMA/Oceananigans.jl/pull/3618 / https://buildkite.com/clima/oceananigans/builds/16126#01903c25-aa5f-4e96-9088-70984b90ebe4 seemingly)

File: ext/EnzymeExt.jl (hunk 1 above)
Comment (vchuravy) [reply ext/EnzymeExt.jl:129]: We should double check that this adapts to what we want on CUDA. I don't know if CuRef is right here...

Comment (wsmoses) [reply ext/EnzymeExt.jl:129]: Sure but this is code wouldn’t affect any currently landed cuda support (this is only reverse mode).

So we should likely explore that in michel’s PR?

Comment (luraess) [reply ext/EnzymeExt.jl:129]: And what about non CUDA GPUs?

**Comment by vchuravy** [comment]
@wsmoses just so that you aware, this is what `Ref` get's converted to for `cudacall`.

https://github.com/JuliaGPU/CUDA.jl/blob/84bb117194d42c5ec58c09e85c312d491fedee68/src/compiler/execution.jl#L165-L175

**Pull request merged**